File: javascript/removeDuplicates.js
'use strict';

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    for (let i = nums.length; i >= 0; i--) {
        const temp = nums.slice(i + 1);
        if (temp.includes(nums[i])) {
            nums.splice(i, 1);
        }
    }
    return nums.length;
}

File: golang/removeDuplicates.go
package solution

func removeDuplicates(nums []int) int {
	length := len(nums)
	if length <= 1 {
		return length
	}

	length = 1;
	for i , v := range nums {
		if i > 0 && nums[length - 1] != v {
			nums[length] = v
			length++
		}
	}
	return length
}